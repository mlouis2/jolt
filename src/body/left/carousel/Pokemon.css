.favoriteIcon img {
     height: 30px;
     float: right;
     margin: 20px;
     width: 30px;
}

.pokemonSprite {
     background-color: black;
     float: left;
     width: 100px;
     height: 100px;
     border-bottom: 5px solid black;
     border-right: 5px solid black;
}
