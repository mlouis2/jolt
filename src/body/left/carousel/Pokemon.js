import React from 'react'

function titleCase(str) {
    return str.charAt(0).toUpperCase() + str.substring(1);
}

function formatNumber(number) {
    let strNumber = number.toString();
    while (strNumber.length < 3) {
       strNumber = "0" + strNumber;
    }
    return "#" + strNumber;
}

function formatTypes(types) {
    return types.map(s => titleCase(s)).join('/');
}

// Try Pokemon(pokemonApi, index)
// pokemonApi has the getter functions and caches things.
function Pokemon(sprite = "https://images-na.ssl-images-amazon.com/images/I/51lh93vBeRL._SY679_.jpg", name = "Pokemon", number = "?", types = "?", description = "...", index) {
    const imgAlt = "Image of the Pokémon " + name + ".";
    return (
        <div key={index} className="pokemon" id={name}>
        <div className="pokemonHeader">
        <img src={sprite} alt={imgAlt}/>
        <div className="pokemonName">
        {name}
        </div>
        <div className="pokemonNumberAndType">
        {number} {types}
        </div>
        </div>
        <div className="pokemonDescription">
        {name} is {number}, a {types.toLowerCase()} Pokémon. {description}
        </div>
        </div>
    );
}

export {
    Pokemon,
    titleCase,
    formatNumber,
    formatTypes,
};
