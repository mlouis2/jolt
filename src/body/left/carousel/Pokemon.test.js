import React from 'react'
import ReactDOM from 'react-dom'
import { Pokemon, titleCase, formatNumber, formatTypes } from './Pokemon'

it('can title case strings', () => {
     expect(titleCase("testingTitleCase")).toEqual("TestingTitleCase");
})

it('can format type numbers', () => {
     expect(formatNumber(20)).toEqual("#020");
     expect(formatNumber(2)).toEqual("#002");
     expect(formatNumber(259)).toEqual("#259");
})

it('can format pokemon types', () => {
     expect(formatTypes(["fire", "water"])).toEqual("Fire/Water");
     expect(formatTypes(["grass"])).toEqual("Grass");
})

it('can create a pokemon', () => {
     const fakePokemon = Pokemon("https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png", "Maddiemon", 1, "Grass/Fighting", "Maddiemon is a grass Pokemon.", 0);
     expect(fakePokemon).toMatchSnapshot()
})
