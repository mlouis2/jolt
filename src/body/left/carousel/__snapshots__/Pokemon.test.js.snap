// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a pokemon 1`] = `
<div
  className="pokemon"
  id="Maddiemon"
>
  <div
    className="pokemonHeader"
  >
    <img
      alt="Image of the Pokémon Maddiemon."
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
    <div
      className="pokemonName"
    >
      Maddiemon
    </div>
    <div
      className="pokemonNumberAndType"
    >
      1
       
      Grass/Fighting
    </div>
  </div>
  <div
    className="pokemonDescription"
  >
    Maddiemon
     is 
    1
    , a 
    grass/fighting
     Pokémon. 
    Maddiemon is a grass Pokemon.
  </div>
</div>
`;
